cmd/metal-api/internal/metal/network.go

```diff
@@ -229,7 +231,33 @@ type Network struct {
 	Underlay                   bool              `rethinkdb:"underlay" json:"underlay"`
 	Shared                     bool              `rethinkdb:"shared" json:"shared"`
 	Labels                     map[string]string `rethinkdb:"labels" json:"labels"`
-	AdditionalAnnouncableCIDRs []string          `rethinkdb:"additionalannouncablecidrs" json:"additionalannouncablecidrs" description:"list of cidrs which are added to the route maps per tenant private network, these are typically pod- and service cidrs, can only be set for private super networks"`
+	AddressFamilies            AddressFamilies   `rethinkdb:"addressfamilies" json:"addressfamilies"`
+	AdditionalAnnouncableCIDRs []string          `rethinkdb:"additionalannouncablecidrs" json:"additionalannouncablecidrs" description:"list of cidrs which are added to the route maps per tenant private network, these are typically pod- and service cidrs, can only be set in a supernetwork"`
+}
+
+type ChildPrefixLength map[AddressFamily]uint8
+
+// AddressFamily identifies IPv4/IPv6
+type AddressFamily string
+type AddressFamilies []AddressFamily
+type AddressFamilyUsage map[AddressFamily]uint64
+
+const (
+	// IPv4AddressFamily identifies IPv4
+	IPv4AddressFamily = AddressFamily("IPv4")
+	// IPv6AddressFamily identifies IPv6
+	IPv6AddressFamily = AddressFamily("IPv6")
+)
+
+// ToAddressFamily will convert a string af to a AddressFamily
+func ToAddressFamily(af string) AddressFamily {
```

Comment:
When returning an error with this function it would be helpful for validations because currently users can often try to send random stuff to the API and try to do something weird.

Reply:
Implemented ValidateAddressfamily as well